File: app/src/main/java/ru/geekbrains/lessions2345/notepadonfragments_2/logic/ListNotes.java
package ru.geekbrains.lessions2345.notepadonfragments_2.logic;

public class ListNotes {
    // Список имён заметок
    private String[] names;
    // Список кратких описаний заметок
    private String[] descriptions;
    // Список дат создания заметок
    private String[] dates;

    public ListNotes(String[] names, String[] descriptions, String[] dates) {
        this.names = names;
        this.descriptions = descriptions;
        this.dates = dates;
    }

    public String getName(int index) {
        return names[index];
    }

    public String getDescription(int index) {
        return descriptions[index];
    }

    public String getDate(int index) {
        return dates[index];
    }

    public int getSize() {
        return names.length;
    }

    public void setName(int index, String name) {
        this.names[index] = name;
    }

    public void setDescription(int index, String description) {
        this.descriptions[index] = description;
    }

    public void setDate(int index, String date) {
        this.dates[index] = date;
    }

    public void addNote(int index, String name, String description, String date) {
        int newArraySize = name.length() + 1;
        if (index < newArraySize) {
            String[] newNames = new String[newArraySize];
            String[] newDescriptions = new String[newArraySize];
            String[] newDates = new String[newArraySize];

            int counter = 0;
            for (int i = 0; i < newArraySize; i++) {
                if (counter == index) {
                    newNames[i] = name;
                    newDescriptions[i] = description;
                    newDates[i] = date;
                } else {
                    newNames[i] = names[counter];
                    newDescriptions[i] = descriptions[counter];
                    newDates[i] = dates[counter];
                    counter++;
                }
            }
        }
    }
}

File: app/src/main/java/ru/geekbrains/lessions2345/notepadonfragments_2/logic/ListNotes.kt
package ru.geekbrains.lessions2345.notepadonfragments_2.logic

class ListNotes (names : Array<String>, descriptions : Array<String>, dates : Array<String>) {
    // Список имён заметок
    private var names: Array<String> = Array<String>(1){""}

    // Список кратких описаний заметок
    private var descriptions: Array<String> = Array<String>(1){""}

    // Список дат создания заметок
    private var dates: Array<String> = Array<String>(1){""}

    init {
        this.names = names
        this.descriptions = descriptions
        this.dates = dates
    }

    fun getName(index: Int): String {
        return names[index]
    }

    fun getDescription(index: Int): String {
        return descriptions[index]
    }

    fun getDate(index: Int): String {
        return dates[index]
    }

    fun getSize(): Int {
        return names.size
    }

    fun setName(index: Int, name: String) {
        names[index] = name
    }

    fun setDescription(index: Int, description: String) {
        descriptions[index] = description
    }

    fun setDate(index: Int, date: String) {
        dates[index] = date
    }

    public fun addNote(index : Int, name : String, description : String, date : String) {
        var newArraySize : Int = this.names.size + 1
        if (index < newArraySize) {
            var newNames : Array<String> = Array<String>(newArraySize){""}
            var newDescriptions : Array<String> = Array<String>(newArraySize){""}
            var newDates : Array<String> = Array<String>(newArraySize){""}

            var counter : Int = 0
            for (i in 0..newArraySize - 1) {
                if (counter == index) {
                    newNames[i] = name
                    newDescriptions[i] = description
                    newDates[i] = date
                } else {
                    newNames[i] = names[counter]
                    newDescriptions[i] = descriptions[counter]
                    newDates[i] = dates[counter]
                    counter++
                }
            }
        }
    }
}